assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js

```diff
@@ -56,125 +60,119 @@ export default function UserDimensionsPieChart( { dimensionName, dimensionValue,
 	const notSetSupportURL = useSelect( ( select ) => select( CORE_SITE ).getGoogleSupportURL( {
 		path: '/analytics/answer/2820717',
 	} ) );
-
-	// Create a unique chartID to use for this component's GoogleChart child component.
-	const chartID = `user-dimensions-pie-chart-${ useInstanceId( UserDimensionsPieChart ) }`;
+	const dimensionColor = useSelect( ( select ) => select( CORE_UI ).getValue( UI_DIMENSION_COLOR ) );
 
 	const { setValues } = useDispatch( CORE_UI );
-	const onReady = useCallback( () => {
-		setChartLoaded( true );
-
-		const chartData = GoogleChart.charts.get( chartID );
-		const { chart, onSelect } = chartData || {};
-		const { slices } = UserDimensionsPieChart.chartOptions;
-
-		if ( chart && ! onSelect ) {
-			chartData.onSelect = global.google.visualization.events.addListener( chart, 'select', () => {
-				const { row } = chart.getSelection()?.[ 0 ] || {};
-				if ( row !== null && row !== undefined ) {
-					const { dataTable } = GoogleChart.charts.get( chartID ) || {};
-					if ( dataTable ) {
-						const newDimensionValue = dataTable.getValue( row, 0 );
-						const isOthers = __( 'Others', 'google-site-kit' ) === newDimensionValue;
-
-						if ( isOthers ) {
-							chart.setSelection( [] );
-						}
 
-						setValues( {
-							[ UI_DIMENSION_VALUE ]: isOthers ? '' : newDimensionValue,
-							[ UI_DIMENSION_COLOR ]: isOthers ? '' : slices[ row ]?.color,
-						} );
-					}
-				} else {
-					setValues( {
-						[ UI_DIMENSION_VALUE ]: '',
-						[ UI_DIMENSION_COLOR ]: '',
-					} );
-				}
+	const chartWrapperRef = useRef();
+
+	const { slices } = UserDimensionsPieChart.chartOptions;
+
+	const onLegendClick = ( index ) => {
+		if ( ! chartWrapperRef.current ) {
+			return;
+		}
+
+		const newDimensionValue = chartWrapperRef.current.getDataTable().getValue( index, 0 );
+		const isOthers = __( 'Others', 'google-site-kit' ) === newDimensionValue;
+
+		if ( isOthers ) {
+			return;
```

Comment:
@tofumatt we probably need to remove this `return` statement because it prevents dimension value reset when "others" is selected in the legend. Currently, if I select a slice, then click on the "others" slice - my selection is reset. But it doesn't happen if I click on the legend item.